objects/src/transaction/executed_tx.rs

```diff
@@ -0,0 +1,92 @@
+use miden_core::StackOutputs;
+
+use super::{utils, Account, AdviceInputs, Digest, Note, StackInputs};
+
+pub struct ExecutedTransaction {
+    initial_account: Account,
+    final_account: Account,
+    consumed_notes: Vec<Note>,
+    created_notes: Vec<Note>,
+    tx_script_root: Option<Digest>,
+    block_ref: Digest,
+}
```

Comment:
At the minute the `ExecutedTransaction` object is primarily inspired by the requirements of this PR.  We need an object that can provide a summary of the full execution cycle of the transaction, including both inputs and outputs.  I think the `PreparedTransaction` serves a slightly different purposes.  My understanding is it enables stateless execution and proving of transactions via a self contained advice provider.  I expect the `ExecutedTransaction` will evolve in the future.

Reply:
At the minute yes.  But the idea is that it's an object that holds all information related to an executed transaction including initial / final account states and consumed / created notes.  I guess you could consider it a `TransactionResult` of sorts.

Reply:
Yes - I think this is fine for now. For the future, I think it might need to change quite a bit. For example:

- For `created_notes` we may not have full note details to instantiate `Note` structs (i.e., we may know only the `recipient` but not specific serial number, inputs, or script root).
- Instead of having fully realized `Account` structs for initial/final state, it may contain account ID, initial/final hash, and some sort of "change descriptor" object which contains the state delta between the two account states.

But as you said, once we do this, it'll probably evolve into `PreparedTransaction` object.